python/dgl/graphbolt/negative_sampler.py

```diff
@@ -69,7 +71,34 @@ def _sample(self, minibatch):
             else:
                 self._collate(minibatch, self._sample_with_etype(node_pairs))
         else:
-            raise NotImplementedError("Not implemented yet.")
+            seeds = minibatch.seeds
+            if isinstance(seeds, Mapping):
+                if minibatch.indexes is None:
+                    minibatch.indexes = {}
+                for etype, pos_pairs in seeds.items():
+                    assert pos_pairs.ndim == 2 and pos_pairs.shape[1] == 2, (
+                        "Only tensor with shape N*2 is supported for negative"
+                        + f" sampling, but got {pos_pairs.shape}."
+                    )
+                    self._collate(
+                        minibatch,
+                        self._sample_with_etype(
+                            pos_pairs, etype, use_seeds=True
+                        ),
+                        etype,
+                    )
+                    minibatch.indexes[etype] = self._construct_indexes(
+                        pos_pairs
+                    )
+            else:
+                assert seeds.ndim == 2 and seeds.shape[1] == 2, (
```

Comment:
Remove this limitation for future expand.

Reply:
Yes, we should move this to UniformNegativeSampler.